File: 80/solution.py
from typing import List


class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        idx = 0
        dup = False
        val = -10 ** 4 - 1
        for i in range(len(nums)):
            if nums[i] == val and dup:
                continue
            elif nums[i] != val:
                dup = False
            else:
                dup = True
            nums[idx] = nums[i]
            val = nums[i]
            idx += 1
        return idx

File: 80/solution.go
package _80

func removeDuplicates(nums []int) int {
	idx := 0
	dup := false
	val := -10001
	for i := 0; i < len(nums); i++ {
		if nums[i] != val {
			nums[idx] = nums[i]
			idx += 1
			val = nums[i]
			dup = false
		} else if dup == false {
			nums[idx] = nums[i]
			idx += 1
			dup = true
		}
	}
	return idx
}
